pages/api/admin/catalogue/index.js

```diff
@@ -17,14 +17,22 @@ export default async function handler(request, response) {
     await dbConnect();
     switch (request.method) {
       case "GET": {
-        const plants = await Plant.find();
+        const plants = await Plant.find().lean();
+
+        if (plants && plants.length > 0) {
+          await Promise.all(
+            plants.map(async (plant) => {
+              if (plant.imageStoragePath) {
+                plant.storedImageUrl = await getSignedImageUrl(
+                  plant.imageStoragePath
+                );
+              }
```

Comment:
please don't mutate the existing plants array but use the return value to map to a new array and send that to your client instead.